src/main/java/com/datapipe/jenkins/vault/model/VaultSecretValue.java

```diff
@@ -51,6 +52,13 @@ public VaultSecretValue(String envVar, @NonNull String vaultKey) {
     @DataBoundConstructor
     public VaultSecretValue(@NonNull String vaultKey) {
         this.vaultKey = fixEmptyAndTrim(vaultKey);
+        this.isRequired = true;
```

Comment:
Why is this here? Not sure this should be the default
```suggestion
```

Reply:
The current default behavior is that all vault secrets are required (which is why I created this PR)... Figured I'd add this default as to not change current behavior, only add an additional option. I personally don't care either way though.

Reply:
This is in reference to the values, not the secrets themselves.
`FailIfNotFound` fails if the secrets are not found
`isRequired` on the secret values is specifically when a secret does exist, but an optional value does not.